src/browser/deck/actions.js

```diff
@@ -214,24 +273,19 @@ define((require, exports, module) => {
   // on a result of this function as they assume to have an active item.
   const deactivate = items => items.update(indexOfActive(items), asInactive);
 
-  const insert = (items, item, index) => {
-    const include = items =>
-      index <= 0 ? items.push(asFirst(items, item)) :
-      index >= items.count() - 1 ? items.push(asLast(items, item)) :
-      items.push(asNth(items, index, item));
-
-    const update = compose(include,
-                           isSelected(item) ? deselect : identity,
-                           isActive(item) ? deactivate : identity);
-
-    return update(items);
+  const include = (items, item) => {
+    const edit = compose(items => items.push(item),
```

Comment:
It's not an array but immutable list, all methods return new lists in this case with item added